app/controllers/trips_controller.rb

```diff
@@ -1,17 +1,33 @@
 class TripsController < ApplicationController
-  before_action :set_trip, only: [:dashboard, :show, :edit, :update, :destroy]
+  before_action :set_trip, only: [:dashboard, :import, :show, :edit, :update, :destroy]
+  skip_before_action :verify_authenticity_token, only: [:import]
 
   # GET /trips
   # GET /trips.json
   def index
     authorize Trip
-    @trips = Trip.all
-  end
-
+    @trips = Trip.all end
   def dashboard
     authorize Trip
   end
 
+  def import
+    if request.post?
+      authorize Trip
+      @trip = Trip.find(params[:id])
+      @data = trip_params[:importStudents].split(" ")
```

Comment:
I would call this array `osises`

Reply:
You seem to do `i.to_i` a lot below. If you want, you can simply turn the entire array of strings into ints using `map`: https://docs.ruby-lang.org/en/2.0.0/Array.html#method-i-map